### PR TITLE
fix: Jump to message only works once

### LINKED ISSUE #31653: dynamic search slug not working properly
### Description:

If we want to search a particular item or word in the chats then we can simply use the search box. On first time entry it works. But on further searching the search urls slug in address bar gets wrongly updated.

### Steps to reproduce:

1. Go to search box on top left.
2. Search any item
3. Click on jump to that message button
4. Then search something else

### Expected behavior:

It should give the exact message in the chats 

### Actual behavior:

Update the address bar slug incorrectly hence no search found further

### Server Setup Information:

- Version of Rocket.Chat Server: 6.4.8
- Operating System: win 10
- Deployment Method:
- Number of Running Instances: 1
- DB Replicaset Oplog: enabled
- NodeJS Version: v14.21.3
- MongoDB Version: 5

### Client Setup Information

- Desktop App or Browser Version: browser
- Operating System: win 10

### Additional context

https://github.com/RocketChat/Rocket.Chat/assets/93787385/7883a912-6c90-4edd-81d2-fe006d5b9984

### PR DESCRIPTION
**Pull request by SySagar**
fix: dynamic slug during search
feat: patch update

 
  - I have read the Contributing Guide - https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat doc
  - I have signed the CLA - https://cla-assistant.io/RocketChat/Rocket.Chat
  - Lint and unit tests pass locally with my changes
  - I have added tests that prove my fix is effective or that my feature works (if applicable)
  - I have added necessary documentation (if applicable)
  - Any dependent changes have been merged and published in downstream modules

## Proposed changes (including videos or screenshots)

Updated the slug update logic in the `setMessageJumpQueryStringParameter` module. Now works good.

## Issue(s)
closes #31653 
[CORE-56](https://rocketchat.atlassian.net/browse/CORE-56)
https://github.com/RocketChat/Rocket.Chat/issues/31727

## Steps to test or reproduce
- Steps to reproduce:
- Go to search box on top left.
- Search any item
- Click on jump to that message button
- Then search something else

[CORE-56]: https://rocketchat.atlassian.net/browse/CORE-56?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by gabriellsh — CHANGES_REQUESTED** [review]
Hey, thanks for the contribution!

I'll rename this PR to a more appropriate name.
I understand the reasoning behind what you did to fix the issue, but I think we can do one better. If you reset the `setMessageJumQueryStringParameter.ts` file, you can do the following:

```
const locationPathname = new URL(window.location.href).pathname as LocationPathname
```

This will have pretty much the same effect, but will be significantly easier to read, and will also prevent us from duplicating other query string parameters.

**Comment by SySagar** [comment]
@gabriellsh I have updated your changes. Please review.

File: apps/meteor/client/lib/utils/setMessageJumpQueryStringParameter.ts [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
-
+import { LocationPathname } from '@rocket.chat/ui-contexts';
 import { router } from '../../providers/RouterProvider';
 
 export const setMessageJumpQueryStringParameter = async (msg: IMessage['_id'] | null) => {
-	const { msg: _, ...search } = router.getSearchParameters();
+	const { msg: existingMsg, ...search } = router.getSearchParameters();
+
+	// Update the 'msg' parameter conditionally
+	const updatedSearch = msg ? { ...search, msg } : search;
```
Comment (gabriellsh) [inline apps/meteor/client/lib/utils/setMessageJumpQueryStringParameter.ts:9]:
```suggestion

```

File: apps/meteor/client/lib/utils/setMessageJumpQueryStringParameter.ts [hunk 2]
```diff
@@ -1,14 +1,19 @@
 import type { IMessage } from '@rocket.chat/core-typings';
-
+import { LocationPathname } from '@rocket.chat/ui-contexts';
 import { router } from '../../providers/RouterProvider';
 
 export const setMessageJumpQueryStringParameter = async (msg: IMessage['_id'] | null) => {
-	const { msg: _, ...search } = router.getSearchParameters();
+	const { msg: existingMsg, ...search } = router.getSearchParameters();
```
Comment (gabriellsh) [inline apps/meteor/client/lib/utils/setMessageJumpQueryStringParameter.ts:6]:
```suggestion
	const { msg: _, ...search } = router.getSearchParameters();
```

File: apps/meteor/client/lib/utils/setMessageJumpQueryStringParameter.ts [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+
+	// Update the 'msg' parameter conditionally
```
Comment (gabriellsh) [inline apps/meteor/client/lib/utils/setMessageJumpQueryStringParameter.ts:8]:
```suggestion
```

**Review by gabriellsh — CHANGES_REQUESTED** [review]

**Comment by SySagar** [comment]
ok then I am reseting back all my commits and updating the changes. I guess this will be ok.

**Review by gabriellsh — APPROVED** [review]

**Comment by SySagar** [comment]
any update for this pr @gabriellsh @dougfabris ?

**Comment by SySagar** [comment]
hey @gabriellsh @dougfabris I have fixed the linting issue. Can you now merge it please?

**Review by yash-rajpal — DISMISSED** [review]
LGTM

**Comment by SySagar** [comment]
hey @yash-rajpal , seems there were some more lint issue. I fixed them up.

File: .changeset/gold-tigers-rhyme.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@rocket.chat/meteor': patch
+---
+
+fix: dynamic slug during search
```
Comment (ggazzo) [inline .changeset/gold-tigers-rhyme.md:5]: please rephrase the changeset  to match the pr title

**Review by ggazzo — CHANGES_REQUESTED** [review]

**Comment by SySagar** [comment]
@ggazzo changesets updated.

File: .changeset/gold-tigers-rhyme.md (hunk 4 above)
Comment (SySagar) [reply .changeset/gold-tigers-rhyme.md]: @ggazzo changesets updated

**Comment by SySagar** [comment]
hey @ggazzo can you please approve this

**Pull request merged**